src/net_processing.cpp

```diff
@@ -597,6 +607,9 @@ struct CNodeState {
     //! Whether this peer relays txs via wtxid
     bool m_wtxid_relay{false};
 
+    //! Whether this peer may be used by us for flooding txs.
+    bool m_we_may_flood_to{false};
```

Comment:
Again, no need for this to be protected by main. I think it can easily live in `Peer`.